sqlite/lib/SQLiteService.js

```diff
@@ -167,7 +167,12 @@ class SQLiteService extends SQLService {
     val(v) {
       if (Buffer.isBuffer(v.val)) v.val = v.val.toString('base64')
       // intercept DateTime values and convert to Date objects to compare ISO Strings
-      else if (/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}[Z+-]/.test(v.val)) v.val = new Date(v.val)
+      else if (/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(.\d{1,9})?(Z|[+-]\d{2}(:?\d{2})?)$/.test(v.val)) {
```

Comment:
Why do we need to intercept that at all?

Reply:
This has to be unified for when selecting data from a `DateTime` columns and then using the result in a `where` clause. As the `DateTime` value will be stored on `SQLite` in `Timestamp` formatting, but will be returned as `DateTime` format when selected.

Reply:
Also this logic is only for `SQLite` as all other database have binary `Date` / `Time` formats.

Reply:
Do we have a test that shows that use case and expected behavior? 
Respectively: we should have one, I think :)

Reply:
Added a test which shows that both `Timestamp` and `DateTime` values can be compared against `DateTime` columns (https://github.com/cap-js/cds-dbs/pull/410/commits/46805e6846562f3272c647909dc255119a81f92b).

Reply:
👍 thanks. 
But then: these tests are also green when I revert that change here

Reply:
Yes the test that check the current change is [this](https://github.com/cap-js/cds-dbs/pull/410/files#diff-a51794365405e10091720b99518753899c1f3d2dd6395edf483b465560c51e9a). Where the string contains a valid `DateTime` / `Timestamp`, but should not be converted into a `Date`.